core/trino-main/src/test/java/io/trino/sql/planner/AbstractPredicatePushdownTest.java

```diff
@@ -589,6 +597,147 @@ WITH t(a) AS (VALUES 'a', 'b')
+
+        // join probe side, max between value pushed down
+        assertPlan(
+                "SELECT p.name, l.comment FROM lineitem l JOIN part p ON l.partkey = p.partkey WHERE l.comment BETWEEN p.name AND 'f'",
+                disableJoinReordering,
+                output(
+                        join(INNER, builder -> builder
+                                .equiCriteria("l_partkey", "p_partkey")
+                                .filter(new Comparison(
+                                        GREATER_THAN_OR_EQUAL,
+                                        new Cast(new Reference(createVarcharType(44), "comment"), createVarcharType(55)),
+                                        new Reference(createVarcharType(55), "name")))
+                                .left(filter(
+                                        new Comparison(
+                                                LESS_THAN_OR_EQUAL,
+                                                new Reference(createVarcharType(44), "comment"),
+                                                new Constant(createVarcharType(44), utf8Slice("f"))),
+                                        tableScan(
+                                                "lineitem",
+                                                ImmutableMap.of("l_partkey", "partkey", "comment", "comment"))))
+                                .right(anyTree(
+                                        tableScan(
+                                                "part",
+                                                ImmutableMap.of("p_partkey", "partkey", "name", "name")))))));
+        // neither side is pushed down
+        assertPlan(
+                "SELECT p.name, l.comment FROM lineitem l JOIN part p ON l.partkey = p.partkey WHERE p.name BETWEEN l.linestatus  AND l.comment",
+                disableJoinReordering,
+                output(
+                        join(INNER, builder -> builder
+                                .equiCriteria("l_partkey", "p_partkey")
+                                .filter(new Between(
+                                        new Reference(createVarcharType(55), "name"),
+                                        new Cast(new Reference(createVarcharType(1), "linestatus"), createVarcharType(55)),
+                                        new Cast(new Reference(createVarcharType(44), "comment"), createVarcharType(55))))
+                                .left(anyIfDynamicFilteringEnabled(tableScan(
+                                        "lineitem",
+                                        ImmutableMap.of("l_partkey", "partkey", "comment", "comment", "linestatus", "linestatus"))))
+                                .right(exchange(tableScan(
+                                        "part",
+                                        ImmutableMap.of("p_partkey", "partkey", "name", "name")))))));
+    }
+
+    private PlanMatchPattern anyIfDynamicFilteringEnabled(PlanMatchPattern source)
```

Comment:
can we just use `any` or `anyTree` or does that result in patten match failing ?

Reply:
I believe both `any` and `anyTree` require at least one node and in this case, there are no nodes for the case without DF